Amazon.IonObjectMapper.Test/CustomIonSerializerTest.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Amazon.IonDotnet;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static Amazon.IonObjectMapper.Test.Utils;
+
+namespace Amazon.IonObjectMapper.Test
+{
+    [TestClass]
+    public class CustomIonSerializerTest
+    {
+        [TestMethod]
+        public void CanUseACustomSerializerFactory()
+        {
+            var customSerializer = new IonSerializer(new IonSerializationOptions { CustomContext = new Dictionary<string, object>() 
+                {
+                    { "customSerializerKey", new CustomSerializerValue()}
+                }
+            });
+            var stream = customSerializer.Serialize(TestObjects.honda);
+            var serialized = StreamToIonValue(stream);
+            var engine = serialized.GetField("engine");
+            
+            Assert.AreEqual(3, engine.GetField("Cylinders").IntValue);
+        }
+
+        [TestMethod]
+        public void CanUseACustomDeserializerFactory()
```

Comment:
This does not use a factory though?

Reply:
This test created `Dictionary<string, object> customContext` which will pass in as a parameter to `IonSerializerFactory` class, and create new custom IonSerializer. 

Changed this test method name to a more expressive name.